src/proxy/ReplicatedMapProxy.ts

```diff
@@ -0,0 +1,201 @@
+import {ReplicatedMapRemoveCodec} from '../codec/ReplicatedMapRemoveCodec';
+import {ReplicatedMapPutAllCodec} from '../codec/ReplicatedMapPutAllCodec';
+import {ReplicatedMapKeySetCodec} from '../codec/ReplicatedMapKeySetCodec';
+import {ReplicatedMapValuesCodec} from '../codec/ReplicatedMapValuesCodec';
+import {ReplicatedMapEntrySetCodec} from '../codec/ReplicatedMapEntrySetCodec';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+import {ReplicatedMapRemoveEntryListenerCodec} from '../codec/ReplicatedMapRemoveEntryListenerCodec';
+import {EntryEventType} from '../core/EntryEventType';
+import ClientMessage = require('../ClientMessage');
+/* tslint:disable:max-line-length */
+import {ReplicatedMapAddEntryListenerToKeyWithPredicateCodec} from '../codec/ReplicatedMapAddEntryListenerToKeyWithPredicateCodec';
+import {ReplicatedMapAddEntryListenerToKeyCodec} from '../codec/ReplicatedMapAddEntryListenerToKeyCodec';
+import {ReplicatedMapAddEntryListenerWithPredicateCodec} from '../codec/ReplicatedMapAddEntryListenerWithPredicateCodec';
+import {ReplicatedMapAddEntryListenerCodec} from '../codec/ReplicatedMapAddEntryListenerCodec';
+/* tslint:enable:max-line-length */
+
+export class ReplicatedMapProxy<K, V> extends BaseProxy implements IReplicatedMap<K, V> {
+
+    put(key: K, value: V, ttl: number): Promise<V> {
+        assertNotNull(key);
+        assertNotNull(value);
+
+        let valueData: Data = this.toData(value);
+        let keyData: Data = this.toData(key);
+
+        return this.encodeInvokeOnKey<V>(ReplicatedMapPutCodec, keyData, keyData, valueData, ttl);
+    }
+
+    clear(): Promise<void> {
+        return this.encodeInvokeOnRandomTarget<void>(ReplicatedMapClearCodec);
+    }
+
+    get(key: K): Promise<V> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<V>(ReplicatedMapGetCodec, keyData, keyData);
+    }
+
+    containsKey(key: K): Promise<boolean> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<boolean>(ReplicatedMapContainsKeyCodec, keyData, keyData);
+    }
+
+    containsValue(value: V): Promise<boolean> {
+        assertNotNull(value);
+
+        const valueData = this.toData(value);
+        return this.encodeInvokeOnRandomPartition<boolean>(ReplicatedMapContainsValueCodec, valueData);
+    }
+
+    size(): Promise<number> {
+        return this.encodeInvokeOnRandomPartition<number>(ReplicatedMapSizeCodec);
+    }
+
+
+    isEmpty(): Promise<boolean> {
+        return this.encodeInvokeOnRandomPartition<boolean>(ReplicatedMapIsEmptyCodec);
+    }
+
+    remove(key: K): Promise<V> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<V>(ReplicatedMapRemoveCodec, keyData, keyData);
+    }
+
+    putAll(pairs: [K, V][]): Promise<void> {
+        let pair: [K, V];
```

Comment:
I've been using `const/let` more than a year in different environments and compiled to es5, and had no issues related to them. But in opposite, I highly recommend to use them to reduce bugs during compilation time, even more - most of IDEs can highlight you dangerous code like assigning new value to the constant.

just made several examples:

```javascript
for (let i = 0; i < 10; i += 1) {
//
}
console.log(i);
```
`npm run compile` => exits with 1 and message "test.ts(4,13): error TS2304: Cannot find name 'i'."

```javascript
let a = 1;
let b = 2;
if (true) {
    let a = 3;
}

console.log(a);
```

`npm run compile` => compiled successfully to

```javascript
var a = 1;
var b = 2;
if (true) {
    var a_1 = 3;
}
console.log(a);
```
as you see variable "a" resolved correctly. 

But in case of using `var` you **won't have any compiler notice** and potentially will have much more hard spotted bugs.